### PR TITLE
Add info about black to code_style.rst

### PR DESCRIPTION
**Pull request by anoadragon453**
Fixes https://github.com/matrix-org/synapse/issues/5533

Adds information about how to install and run `black` on the codebase.

### CODE REVIEW

**Comment by anoadragon453** [comment]
No worries, that'd just be some deleting :)

File: docs/code_style.rst [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+formatted.
+
+First install ``black`` with::
+
+  pip install --upgrade black
+
+Have ``black`` auto-format your code (it shouldn't change any
+functionality) with::
+
+  black --check . --exclude="\.tox|build|env"
```
Comment (richvdh) [inline docs/code_style.rst]: doesn't --check just tell you if it's wrong, rather than formatting it for you?

File: docs/code_style.rst [hunk 2]
```diff
@@ -1,5 +1,62 @@
...
+
+  ``flake8`` is a code checking tool. We require code to pass ``flake8`` before being merged into the codebase.
+
+  Install ``flake8`` with::
+
+    pip install --upgrade flake8
+
+  Check all application and test code with::
+
+    flake 8 synapse tests
```
Comment (richvdh) [inline docs/code_style.rst]:
```suggestion
    flake8 synapse tests
```

File: docs/code_style.rst [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+
+It's worth noting that modern IDEs and text editors can run these tools
+automatically on save. It may be worth looking into whether this
+functionality is supported in your editor for a more convenient development
+workflow. It is not, however, recommended to run ``flake8`` on save as it
+takes a while and is very resource intensive.
+
+## General rules
+
+These rules are useful to keep in mind while programming, but be aware that the above tools will handle most of this for you
```
Comment (richvdh) [inline docs/code_style.rst]: I'd get rid of the bits that black handles  here (indenting, line length, blank lines, whitespace). The indenting section is plain wrong and the others are incomplete and they also distract from the bits that still have value (naming, comments, imports).

File: docs/code_style.rst [hunk 4]
```diff
@@ -1,70 +1,69 @@
...
+functionality is supported in your editor for a more convenient development
+workflow. It is not, however, recommended to run ``flake8`` on save as it
+takes a while and is very resource intensive.
 
-- **Whitespace**:
+## General rules
 
-  There should be spaces where spaces should be and not where there shouldn't
-  be:
+These rules are useful to keep in mind while programming, but be aware that the above tools will handle most of this for you
```
Comment (richvdh) [inline docs/code_style.rst]: no they won't.

**Comment by anoadragon453** [comment]
Thanks for reviewing :D

**Pull request merged**